icu4c/source/test/intltest/caltest.cpp

```diff
@@ -5354,6 +5355,18 @@ void CalendarTest::TestLimitsOrdinalMonth() {
     }
 }
 
+void CalendarTest::TestMaxActualLimitsWithoutGet23006() {
+    UErrorCode status = U_ZERO_ERROR;
+    GregorianCalendar gc(status);
+    gc.set(2025, UCAL_AUGUST, 8);
+    LocalPointer<Calendar> cal(Calendar::createInstance(Locale("en@calendar=chinese"), status), status);
+    cal->setTime(gc.getTime(status), status);
+    int32_t beforeCallingGet = cal->getActualMaximum(UCAL_DAY_OF_MONTH, status);
+    cal->get(UCAL_DAY_OF_MONTH, status);
+    int32_t afterCallingGet = cal->getActualMaximum(UCAL_DAY_OF_MONTH, status);
+    assertEquals("getActualMaximum() should return same value before/after calling get()",
+                 beforeCallingGet, afterCallingGet);
+}
```

Comment:
Is `getActualMaximum()` the only public function on this calendar that's changed its behavior?  This test seem to test the leap-month case, as well as checking that you get the right thing whether or not you've done a `get()` on the `UCAL_DAY_OF_MONTH` field, but do you have existing test to verify that this function still works correctly for non-leap months?

I'm guessing there's another `getMaximum()` method that returns the absolute maximum for the dat-of-month field, that it was already returning the right thing, and that there are already tests for it?

Reply:
I added more test cases